sdk/src/main/java/com/qonversion/android/sdk/internal/userProperties/UserPropertiesServiceImpl.kt

```diff
@@ -0,0 +1,40 @@
+package com.qonversion.android.sdk.internal.userProperties
+
+import com.qonversion.android.sdk.internal.common.BaseClass
+import com.qonversion.android.sdk.internal.common.mappers.UserPropertiesMapper
+import com.qonversion.android.sdk.internal.exception.ErrorCode
+import com.qonversion.android.sdk.internal.exception.QonversionException
+import com.qonversion.android.sdk.internal.logger.Logger
+import com.qonversion.android.sdk.internal.networkLayer.apiInteractor.ApiInteractor
+import com.qonversion.android.sdk.internal.networkLayer.dto.Response
+import com.qonversion.android.sdk.internal.networkLayer.requestConfigurator.RequestConfigurator
+
+internal class UserPropertiesServiceImpl(
+    private val requestConfigurator: RequestConfigurator,
+    private val apiInteractor: ApiInteractor,
+    private val mapper: UserPropertiesMapper,
+    logger: Logger
+) : UserPropertiesService, BaseClass(logger) {
+    override suspend fun sendProperties(properties: Map<String, String>): List<String> {
+        val request = requestConfigurator.configureUserPropertiesRequest(properties)
+        val response = apiInteractor.execute(request)
+
+        return if (response !is Response.Success) {
+            logger.error("propertiesRequest ended with an error. Response code: ${response.code}")
+            emptyList()
+        } else {
+            mapProcessedProperties(response)
+        }
```

Comment:
In this case the controller won't be able to detect the way the request was completed successfully or not. I guess we should throw the exception here to catch and handle it in controller